public/app/plugins/datasource/cloudwatch/config_ctrl.ts

```diff
@@ -36,4 +40,22 @@ export class CloudWatchConfigCtrl {
     { name: 'Monthly', value: 'Monthly', example: '[logstash-]YYYY.MM' },
     { name: 'Yearly', value: 'Yearly', example: '[logstash-]YYYY' },
   ];
+
+  regions = [];
+
+  getRegions() {
+    this.datasourceSrv
+      .loadDatasource(this.current.name)
+      .then(ds => {
+        return ds.getRegions();
```

Comment:
When creating a new datasource, this request takes about 12 seconds for me when I don't have added any valid credentials. During the time the region drop down is empty. So maybe a good idea to still have the client side regions list as well until regions have loaded from backend even if that's not optimal from a maintainability perspective.

Think this is even worse since we change how adding datasources works (in master), i.e. we're saving a datasource directly with only the datasource and type set.

Do you see any other way forward than putting back default regions list in client side?

Reply:
hmm, I don't have good idea. Just revert hard coded region list.